microsoft/adaptivecards-templates
# Shared View Page

This PR lets you create shared views of your template. The shared view URL is `{baseURL}/preview/{templateID}/{version}`. For example, the URL can be something like `http://localhost:3000/preview/5e574c4e04a0d8004e0076c3/1.0`.

## Work done

- created a new action in search reducer to toggle visibility of search bar so that it does not show up on shared view page
- reformatted the router definition in App.tsx so that the sidebar does not show up in the shared view page
- created UI for shared view page
- updated AdaptiveCard.tsx so that it can accept an optional prop `version` so that we can specify what version of a template we want to render. if `version` is not specified the latest version of the template is rendered.
- implemented the copy JSON button on Shared View page

## Screenshot

![image](https://user-images.githubusercontent.com/13890126/75416897-7f50a200-58e4-11ea-9972-2a3f7a161caf.png)

## Steps for testing

- Create a new card using the Designer tool in the app
- Save it
- Use postman to find the templateID of the template you created
- Go to URL `http://localhost:3000/preview/{templateID}/1.0
- Press Copy JSON button and paste somewhere to see if it works

## Next steps

- Need to make this page viewable only if this template has been shared by the owner
- Need to make the sharing modal so that owners can share templates
